packages/client-core/src/user/components/UserMenu/menus/AvatarSelectMenu2.tsx

```diff
@@ -0,0 +1,237 @@
+  const userAvatarMutation = useMutation(userAvatarPath)
+
+  const avatarsData = useFind(avatarPath, {
+    query: {
+      name: {
+        $like: `%${search.query.value}%`
+      },
+      $skip: page.value * AVATAR_PAGE_LIMIT,
+      $limit: AVATAR_PAGE_LIMIT
+    }
+  }).data
+  const currentAvatar = avatarsData.find((item) => item.id === selectedAvatarId.value)
+  const searchTimeoutCancelRef = useRef<(() => void) | null>(null)
+
+  AuthService.useAPIListeners()
+
+  const handleConfirmAvatar = async () => {
+    if (userAvatarId !== selectedAvatarId.value) {
+      const selfAvatarEntity = AvatarComponent.getSelfAvatarEntity()
+      if (!selfAvatarEntity || !hasComponent(selfAvatarEntity, SpawnEffectComponent)) {
+        await userAvatarMutation.patch(null, { avatarId: selectedAvatarId.value }, { query: { userId } })
+        if (selfAvatarEntity) avatarLoading.set(true)
+        else PopupMenuServices.showPopupMenu()
+      }
+    }
+    selectedAvatarId.set('' as AvatarID)
+  }
+
+  const handleSearch = async (searchString: string) => {
+    search.local.set(searchString)
+
+    if (searchTimeoutCancelRef.current) {
+      searchTimeoutCancelRef.current()
+    }
+
+    const debouncedSearchQuery = debounce(() => {
+      search.query.set(searchString)
+    }, 500)
+
+    debouncedSearchQuery()
+
+    searchTimeoutCancelRef.current = debouncedSearchQuery.cancel
+  }
+
+  useEffect(() => {
+    if (avatarLoading.value && isUserReady.value) {
+      avatarLoading.set(false)
+      PopupMenuServices.showPopupMenu()
+    }
+  }, [isUserReady, avatarLoading])
+
+  useEffect(() => {
+    const userAvatar = avatarsData.find((item) => item.id === userAvatarId)
+    if (!currentAvatar && !selectedAvatarId.value && userAvatar) {
+      selectedAvatarId.set(userAvatar?.id)
+    }
+  }, [avatarsData, selectedAvatarId, currentAvatar, userAvatarId])
+
+  const debouncedSearchQueryRef = useRef<ReturnType<typeof setTimeout>>()
+
+  useEffect(() => clearTimeout(debouncedSearchQueryRef.current), [])
+
+  return (
+    <div className="fixed top-0 z-[35] flex h-[100vh] w-full bg-[rgba(0,0,0,0.75)]">
+      <Modal
+        id="select-avatar-modal"
+        className="min-w-34 pointer-events-auto m-auto flex h-[95vh] w-[70vw] rounded-xl [&>div]:flex [&>div]:h-full [&>div]:max-h-full [&>div]:w-full  [&>div]:flex-1 [&>div]:flex-col"
+        title={t('user:avatar.titleSelectAvatar')}
+        onClose={() => PopupMenuServices.showPopupMenu()}
+        showCloseButton={false}
+        onSubmit={handleConfirmAvatar}
+        hideFooter={true}
+        hideHeaderCloseButton={true}
+        rawChildren={
+          <div className="flex h-full w-full flex-1 flex-col">
+            <div className="grid h-full w-full flex-1 grid-cols-[60%,40%] gap-6 px-10 py-2">
+              <div className={`${styles.avatarPreview} bg-gradient-to-b from-[#162941] to-[#114352]`}>
+                <div className={`${styles.stars}`}></div>
```

Comment:
lets use only tailwind styles

Reply:
I can use tailwind for most of the code but there are properties in the code below that cannot be converted to inline tailwind classes such as ` box-shadow`: 

```
stars {
  position: absolute;
  top: 0;
  left: 0;
  width: 2px;
  height: 2px;
  background: transparent;
  box-shadow: 
    10rem 5rem 0 rgb(255 255 255 / 80%),
    15rem 8rem 0 rgb(255 255 255 / 50%),
    18rem 10rem 0 rgb(255 255 255 / 70%),
    12rem 12rem 0 rgb(255 255 255 / 90%),
    5rem 13rem 0 rgb(255 255 255 / 40%),
    25rem 6rem 2px rgb(255 255 255 / 80%),
    30rem 10rem 3px rgb(255 255 255 / 60%),
    35rem 12rem 2px rgb(255 255 255 / 90%),
    40rem 14rem 3px rgb(255 255 255 / 50%),
    45rem 15rem 2px rgb(255 255 255 / 70%),
    8rem 6rem 0 rgb(255 255 255 / 70%),
    20rem 4rem 2px rgb(255 255 255 / 80%),
    28rem 7rem 0 rgb(255 255 255 / 50%),
    33rem 9rem 1px rgb(255 255 255 / 90%),
    38rem 13rem 2px rgb(255 255 255 / 60%);

  @keyframes twinkling {
    0% {
      opacity: 0.6;
    }

    50% {
      opacity: 1;
    }

    100% {
      opacity: 0.6;
    }
  }

  animation: twinkling 5s alternate infinite; 
}

```